learn/getting_started/quick_start.mdx

```diff
@@ -45,27 +47,42 @@ Database path:       "./data.ms"
 Server listening on: "localhost:7700"
 ```
 
-Congratulations! You're ready to move on to the next step!
+You now have a Meilisearch instance running in your terminal window. Keep this window open for the rest of this tutorial.
 
-## Add documents
+This command uses the `--master-key` configuration option to secure Meilisearch. This tutorial uses `aSampleMasterKey` as a master key, but you may change it to any alphanumeric string with 16 or more characters.
 
-For this quick start, we will be using a collection of movies as our dataset. To follow along, first click this link to download the file: <ButtonLink as="a" id="downloadMovie" href="/movies.json" download="movies.json">movies.json</ButtonLink>. Then, move the downloaded file into your working directory.
+Setting a master key is optional but strongly recommended in development environments. Master keys are mandatory in production environments. To learn more about securing Meilisearch, refer to the [API keys guide](/learn/security/master_api_keys).
 
-Open a new terminal window and run the following command:
+## Add documents
 
-<CodeSamples id="getting_started_add_documents_md" />
+In this quick start, you will search through a collection of movies.
 
-Meilisearch stores data in the form of discrete records, called [documents](/learn/core_concepts/documents). Documents are grouped into collections, called [indexes](/learn/core_concepts/indexes).
+To follow along, first click this link to download the file: <ButtonLink as="a" id="downloadMovie" href="/movies.json" download="movies.json">movies.json</ButtonLink>. Then, move the downloaded file into your working directory.
 
 <Capsule intent="note">
-Meilisearch currently only accepts data in JSON, NDJSON, and CSV formats. You can read more about this in our [documents guide](/learn/core_concepts/documents#dataset-format).
+Meilisearch accepts data in JSON, NDJSON, and CSV formats.
 </Capsule>
 
-The previous command added documents from `movies.json` to a new index called `movies` and set `id` as the primary key. If it isn't set manually, Meilisearch [infers](/learn/core_concepts/primary_key#meilisearch-guesses-your-primary-key) it from your dataset.
+Run the following command, opening a new terminal window if you are following this tutorial using `cURL`:
+
+<CodeSamples id="getting_started_add_documents_md" />
+
+Meilisearch stores data in the form of discrete records, called [documents](/learn/core_concepts/documents). Each document is composed of multiple fields, which are pairs of one attribute and one value:
+
+```json
+{
+  "id": 0,
+  "attribute": "value"
+}
+```
+
+Documents are grouped into collections, called [indexes](/learn/core_concepts/indexes).
 
 Every index must have a [primary key](/learn/core_concepts/primary_key#primary-field), an attribute shared across all documents in that index. If you try adding documents to an index and even a single one is missing the primary key, none of the documents will be stored.
```

Comment:
WDYT about putting all of these into a Note capsule? (could be titled "Documents, indexes, and the primary key"

Reasoning: the info-dumping here is important, but it also really breaks up the flow of the guide, with the paragraph on line 83 referring to a "previous command" on line 68, which may also have been run a long time ago depending on whether the reader follows links to learn more about documents, indexes, and/or primary keys. Would be better to group them into more of a parenthetical.